test/unit/local/steps/linux_producer.js

```diff
@@ -0,0 +1,173 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const os = require('os')
+const fs = require('fs')
+const fse = require('fs-extra')
+const path = require('path')
+const { onPlatform } = require('../../../support/helpers/platform')
+const LinuxProducer = require('../../../../core/local/steps/linux_producer')
+
+onPlatform('linux', () => {
+  describe('core/local/steps/linux_producer', () => {
+    describe('API of the producer', () => {
+      let syncPath
+      let producer
+
+      beforeEach(() => {
+        syncPath = fs.mkdtempSync(path.join(os.tmpdir(), 'foo-'))
+        producer = new LinuxProducer({
+          syncPath
+        })
+      })
+
+      it('should register on start and dispose on stop the watcher', async () => {
+        await producer.start()
+        should(producer.watcher).be.not.null()
+        await producer.stop()
+        should(producer.watcher).be.null()
+      })
+
+      it('should change absolute path to relative with process', async () => {
+        const dirname = 'foobar'
+        const batch = [
+          {
+            action: 'created',
+            kind: 'directory',
+            path: path.join(syncPath, dirname)
+          }
+        ]
+        producer.process(batch.map(evt => Object.assign({}, evt)))
+        should(await producer.buffer.pop()).eql(
+          batch.map(evt => Object.assign({}, evt, { path: dirname }))
+        )
+      })
+
+      it('should scan a subfolder tree', async () => {
+        const mypath = ['foo', 'bar', 'baz']
+        mkdirsSync(path.join(syncPath, ...mypath))
+        await producer.start()
+
+        const batches = [
+          {
+            batch: await producer.buffer.pop(),
+            path: path.join(mypath[0])
+          },
+          {
+            batch: await producer.buffer.pop(),
+            path: path.join(mypath[0], mypath[1])
+          },
+          {
+            batch: await producer.buffer.pop(),
+            path: path.join(mypath[0], mypath[1], mypath[2])
+          }
+        ]
+
+        batches.forEach(({ batch, path: originalPath }) => {
+          should(
+            batch.map(({ action, kind, path }) => ({
+              action,
+              kind,
+              path
+            }))
+          ).eql([
+            {
+              action: 'scan',
+              kind: 'unknown',
+              path: originalPath
+            }
+          ])
+        })
+      })
+    })
+
+    describe('register an event on FS events', () => {
+      let syncPath
+      let producer
+
+      before(async () => {
+        syncPath = fs.mkdtempSync(path.join(os.tmpdir(), 'foo-'))
+        producer = new LinuxProducer({
+          syncPath
+        })
+        await producer.start()
+        await producer.buffer.pop()
+      })
+
+      it('detect events on folder in temp dir', async () => {
+        const dirname = 'foobaz'
+        const newname = 'foobarbaz'
+        fs.mkdirSync(path.join(syncPath, dirname))
```

Comment:
Here, `mkdirSync` from `fs` is used, but at other places it's `mkdirsSync` from `fs-extra` that is used. Is there a reason for not using the same everywhere?

Reply:
yes, `mkdirSync` is used to create a single subfolder (`mkdirSync('/temp/foo-xxx', 'foobar')`) when `mkdirsSync` is used to create a suite of subfolders given a path (`mkdirsSync('/temp/foo-xxx', 'foo/bar/baz')`).
The recursive `mkdirSync` will be available in nodejs 10.2.x.
I did write a simple function to compensate for the lack, but @taratatach asked me to use `fs-extra.mkdirsSync`.